sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -29,9 +29,637 @@ public class UnusedPrivateMemberTest
+        public static int PublicStaticMethod() { return 0; } // Noncompliant
+    }
+}
+
+public class NonPrivateMembers
+{
+    internal int InternalMethod() { return 0; }
+    protected int ProtectedMethod() { return 0; }
+    protected internal int ProtectedInternalMethod() { return 0; }
+    public int PublicMethod() { return 0; }
+    internal static int InternalStaticMethod() { return 0; }
+    protected static int ProtectedStaticMethod() { return 0; }
+    protected internal static int ProtectedInternalStaticMethod() { return 0; }
+    public static int PublicStaticMethod() { return 0; }
+
+    public class InnerPublicClass
+    {
+        internal int InternalMethod() { return 0; }
+        protected int ProtectedMethod() { return 0; }
+        protected internal int ProtectedInternalMethod() { return 0; }
+        public int PublicMethod() { return 0; }
+        internal static int InternalStaticMethod() { return 0; }
+        protected static int ProtectedStaticMethod() { return 0; }
+        protected internal static int ProtectedInternalStaticMethod() { return 0; }
+        public static int PublicStaticMethod() { return 0; }
+    }
+}
+
+public interface IInterface
+{
+    int InterfaceMethod();
+}
+
+public class InterfaceImpl : IInterface
+{
+    int IInterface.InterfaceMethod() => 0;
+}
+", new UnusedPrivateMember());
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void Methods_DirectReferences()
+        {
+            Verifier.VerifyCSharpAnalyzer(@"
+using System;
+using System.Linq;
+public class MethodUsages
+{
+    private int Method1() { return 0; }
+    private int Method2() { return 0; }
+    private int Method3() { return 0; }
+    private int Method4() { return 0; }
+    private int Method5() { return 0; }
+    private int Method6() { return 0; }
+    private int Method7() { return 0; }
+    public int Test1(MethodUsages other)
+    {
+        int i;
+        i = Method1();
+        i = this.Method2();
+        Console.Write(Method3());
+        new MethodUsages().Method4();
+        Func<int> x = () => Method5();
+        other.Method6();
+        return Method7();
+    }
+
+    private int Method8() { return 0; }
+    public int ExpressionBodyMethod() => Method8();
+
+    private static int Method9() { return 0; }
+    public MethodUsages(int number) { }
+    public MethodUsages() : this(Method9()) { }
+
+    private int Method10() { return 0; }
+    private int Method11() { return 0; }
+    public object Test2()
+    {
+        var x = new[] { Method10() };
+        var name = nameof(Method11);
+        return null;
+    }
+
+    private int Method12(int i) { return 0; }
+    public void Test3()
+    {
+        new[] { 1, 2, 3 }.Select(Method12);
+    }
+}
+", new UnusedPrivateMember());
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void DebuggerDisplay_Attribute()
```

Comment:
You've labelled one of the regression tests with the related bug number, which is really useful when reviewing and for traceability later. Could you tag the tests for the other tickets with the related bug number too please?

This one is https://github.com/SonarSource/sonar-csharp/issues/1195